contracts/UtilityBrandedToken.sol

```diff
@@ -0,0 +1,265 @@
+        string _symbol,
+        string _name,
+        uint8 _decimals,
+        OrganizationIsWorkerInterface _organization
+    )
+        public
+        Internal(_organization)
+        EIP20Token(_symbol, _name, _decimals)
+    {
+        require(
+            address(_token) != address(0),
+            "Token address is null."
+        );
+        valueToken = _token;
+    }
+
+
+    /* Public functions */
+
+    /**
+     * @notice Public function transfer.
+     *
+     * @dev Function requires that _to address is an internal actor.
+     *
+     * @param _to Address to which BT needs to transfer.
+     * @param _value Number of BTs that needs to transfer.
+     *
+     * @return Success/failure status of transfer.
+     */
+    function transfer(
+        address _to,
+        uint256 _value
+    )
+        public
+        returns (bool)
+    {
+        require(
+            isInternalActor[_to],
+            "To address is not an internal actor."
+        );
+
+        return super.transfer(_to, _value);
+    }
+
+    /**
+     * @notice Public function transferFrom.
+     *
+     * @dev Function requires that _to address is an internal actor.
+     *
+     * @param _from Address from which BT needs to transfer.
+     * @param _to Address to which BT needs to transfer.
+     * @param _value Number of BTs that needs to transfer.
+     *
+     * @return Success/failure status of transferFrom.
+     */
+    function transferFrom(
+        address _from,
+        address _to,
+        uint256 _value
+    )
+        public
+        returns (bool)
+    {
+        require(
+            isInternalActor[_to],
+            "To address is not an internal actor."
+        );
+
+        return super.transferFrom(_from, _to, _value);
+    }
+
+    /**
+     * @notice Public function approve.
+     *
+     * @dev It only allows approval to internal actors.
+     *
+     * @param _spender Address to which msg.sender is approving.
+     * @param _value Number of BTs to be approved.
+     *
+     * @return Success/failure status of approve.
+     */
+    function approve(
+        address _spender,
+        uint256 _value
+    )
+        public
+        returns (bool)
+    {
+        require(
+            isInternalActor[_spender],
+            "Spender is not an internal actor."
+        );
+
+        return super.approve(_spender, _value);
+    }
+
+    /**
+     * @notice Public function mint.
+     *
+     * @dev Function requires:
+     *          - It is called by only CoGateway contract.
+     *          - It only mints to internal actors.
+     *
+     * @param _beneficiary Address of beneficiary.
+     * @param _amount Amount of utility tokens to mint.
+     *
+     * @return True if mint is successful, false otherwise.
+     */
+    function mint(
+        address _beneficiary,
+        uint256 _amount
+    )
+        public
+        onlyCoGateway
+        returns (bool)
+    {
+        require(
+            (isInternalActor[_beneficiary]),
```

Comment:
If I understand correctly, this requirement means that the facilitator must be listed as an internal actor for CoGateway minting to succeed. But I also understand that facilitators are an open/unknown set. 

The requirement for unknown facilitators appears inconsistent with requiring a known facilitator to complete the process, especially if each minting facilitator needs to be made known to each individual UBT.

Or is it the case that minting is not meant to be limited to internal actors, only transferring—a minting facilitator is rewarded in minted UBT, but her only options are to hold or redeem, unless she _also happens to be_ an internal actor? If this is the case, does this mean we are requiring economy owners to be OK with some portion of the tokens they mint not circulating through the economy?

Please clarify for me if I have misunderstood something critical to this point; if I have not, please confer with (at least) @deepesh-kn / @sarvesh-ost to determine the path to resolve this inconsistency.

Reply:
Facilitator will get the reward in branded token.So, facilitator needs to be registered internal actor for EIP20 CoGateway to succeed.